etcdserver/server.go

```diff
@@ -969,6 +969,9 @@ func (s *EtcdServer) Stop() {
 		plog.Warningf("%s failed to transfer leadership (%v)", s.ID(), err)
 	}
 	s.HardStop()
+	if s.authStore != nil {
+		s.authStore.AuthDisable()
```

Comment:
This isn't a good way to clean up goroutines associated with `authStore`; `AuthDisable` is meant to save to disk that auth is disabled; on restart it'll still be disabled. This may coincidentally appear to work because the backend is closed by this point so it won't accept the write, but it seems like it should still panic from misuse...

The `AuthStore` interface probably needs a `Close() error` function to handle cleanup instead of trying to reuse `AuthDisable` and potentially clobbering the auth settings.